### ISSUE TITLE
Get different results with the same seed

### ISSUE THREAD

**Issue by invitdia** [ZhaofengWu/SIFT#4]
Hi, Zhaofeng, thank you for your brilliant work!

I am stuck by a problem when I use your model for training. I ran this model multiple times with the same seed and parameters for the task CoLA, but the best_mcc results did not remain unchanged like 63.1 for the first time and 62.8 for the second time.

May I ask whether you encountered the same issue before or any suggestions would be greatly appreicated!

Thanks.

**Comment by ZhaofengWu** [comment]
I don't recall seeing that issue, but maybe try setting the flags suggested here https://pytorch.org/docs/stable/notes/randomness.html?

**Comment by invitdia** [comment]
Thanks for your link. But it seems that those strategies do not work for me. I have one additional question about the results in the table. Did you use the same set of parameters for all seeds or you adjust parameters for every seed?

**Comment by ZhaofengWu** [comment]
The hyperparameters are searched for each dataset individually. For the bounds see Appendix B. For specific settings, I think each model has a `args.json` file that lists the values.

**Comment by invitdia** [comment]
Thanks for your reply and sorry for bothering again. But I have question about your results. According to my understanding, you searched the best hyperparameters for each dataset and ran the model with the best parameters on each dataset for several times. Finally, you got your results by averaging those outputs. Am I right about your process? Please point out if I make mistakes. Thank you.

**Comment by ZhaofengWu** [comment]
Yes

**Comment by invitdia** [comment]
Thank you for your reply. However, after getting the best parameters, did you run the model for several times with the same seed or run the model several times with different seeds? For example, for the best parameters of task CoLA, did you run the model several times with the same seed 101 or you run it several times with seeds 101, 100 ,102?  Btw, I still can not ensure reproducing the same results if I run your code many times for the same seed.

**Comment by ZhaofengWu** [comment]
We ran with different seeds.

**Comment by ZhaofengWu** [comment]
If you're getting different results, and consistency is important for your use case, I suggest inspecting the tensors after each operation and see where the differences come from.
